ambry-cloud/src/main/java/com/github/ambry/cloud/azure/AzureStorageCompactor.java

```diff
@@ -254,16 +253,10 @@ private int compactPartitionBucketed(String partitionPath, String fieldName, lon
    * @param endTime the end of the query time range.
    * @param maxEntries the max number of metadata records to return.
    * @return a List of {@link CloudBlobMetadata} referencing the deleted blobs found.
-   * @throws CloudStorageException if the query fails.
    */
   List<CloudBlobMetadata> getDeadBlobs(String partitionPath, String fieldName, long startTime, long endTime,
-      int maxEntries) throws CloudStorageException {
-    try {
-      return cosmosDataAccessor.getDeadBlobs(partitionPath, fieldName, startTime, endTime, maxEntries);
-    } catch (DocumentClientException dex) {
-      throw AzureCloudDestination.toCloudStorageException(
-          "Failed to query deleted blobs for partition " + partitionPath, dex, azureMetrics);
-    }
+      int maxEntries) {
+    return cosmosDataAccessor.getDeadBlobs(partitionPath, fieldName, startTime, endTime, maxEntries);
```

Comment:
No exception check needed here?

Reply:
Thank you for catching. Added the exception check.